framework/helpers/BaseFileHelper.php

```diff
@@ -462,7 +462,9 @@ public static function createDirectory($path, $mode = 0775, $recursive = true)
             $result = mkdir($path, $mode);
             chmod($path, $mode);
         } catch (\Exception $e) {
-            throw new \yii\base\Exception("Failed to create directory '$path': " . $e->getMessage(), $e->getCode(), $e);
+            if(!is_dir($path)){
+                throw new \yii\base\Exception("Failed to create directory '$path': " . $e->getMessage(), $e->getCode(), $e);
+            }
```

Comment:
I was under the conception that it was returning the file path, but now I see it is returning the $result. I will upload the fix.